main.go

```diff
@@ -0,0 +1,47 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+package main
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/internal"
+)
+
+func runCmd(args ...string) error {
+	path := args[0]
+	args = args[1:]
+	cmd := exec.Command(path, args...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func isCompilePackage(args []string, pkg string) bool {
+	checks := []string{"compile", "-buildid", "-p " + pkg}
+	for _, check := range checks {
+		if !strings.Contains(strings.Join(args, " "), check) {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
This I would do somewhat differently... In essence the only thing that is relevant here is:
- `argv[0]` ends in `compile` (or `compile.exe` on Windows)
- `argv[1:]` does not include `-V=full` (alternatively, there is `-buildid` and `-p`)

The repeated call to `strings.Join` here is not ideal (and joining should not be necessary at all)